pygmt/src/project.py

```diff
@@ -222,13 +222,18 @@ def project(data=None, x=None, y=None, z=None, outfile=None, **kwargs):
             "The `convention` parameter is not allowed with `generate`."
         )
 
+    # z is optional
+    vectors, names = [x, y], "xy"
+    if z is not None:
+        vectors.append(z)
```

Comment:
Need to append 'z' to `names` here? Also, need a unit test for this if-condition.

Reply:
Actually no. The problem is that `project` only requires two columns, but three or more columns are required. Currently, the variable `names` are used for two purposes: (1) names of passed columns; (2) the number of columns. So, if we append `z` to `names` here, the calling `pygmt.project(data=data)` will fail if `data` has only two columns. I think we still need to maintain a variable for the number of required columns.